include/generic_sort.h

```diff
@@ -31,7 +31,7 @@ Sort make_generic_sort(SortKind sk, Sort sort1);
 Sort make_generic_sort(SortKind sk, Sort sort1, Sort sort2);
 Sort make_generic_sort(SortKind sk, Sort sort1, Sort sort2, Sort sort3);
 Sort make_generic_sort(SortKind sk, SortVec sorts);
-
+Sort make_generic_sort(Datatype dt, std::string & name);
```

Comment:
```suggestion
Sort make_generic_sort(Datatype dt);
```
(and propagate to other places where appropriate)